PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:16 + add:0, change matches:
-    _inherit = ['base.partner.merge.automatic.wizard']
-    _inherit = ['ir.asset']
-    _inherit = ["ir.attachment"]
-    _inherit = ['ir.model.data']
-    _inherit = ['ir.rule']
-    _inherit = ['ir.ui.menu']
-    _inherit = ["res.company"]
-    _inherit = ['res.config.settings']
-    _inherit = ["res.lang"]
-    _inherit = ['res.users']
-    _inherit = ['ir.ui.view']
-    _inherit = ['ir.asset']
-    _inherit = ['ir.attachment']
-    _inherit = ['website.menu']
-    _inherit = ['website.page']
-    _inherit = ['website']

Total Changes: 71

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/website/models/assets.py
+++ b/addons/website/models/assets.py
@@ -10,7 +10,7 @@ from odoo import api, models
 
 
 class Web_EditorAssets(models.AbstractModel):
-    _inherit = ['web_editor.assets']
+    _inherit = 'web_editor.assets'
 
     @api.model
     def make_scss_customization(self, url, values):

--- a/addons/website/models/base_partner_merge.py
+++ b/addons/website/models/base_partner_merge.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class BasePartnerMergeAutomaticWizard(models.TransientModel):
-    _inherit = ['base.partner.merge.automatic.wizard']
+    _inherit = 'base.partner.merge.automatic.wizard'
 
     @api.model
     def _update_foreign_keys(self, src_partners, dst_partner):

--- a/addons/website/models/ir_asset.py
+++ b/addons/website/models/ir_asset.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class IrAsset(models.Model):
-    _inherit = ['ir.asset']
+    _inherit = 'ir.asset'
 
     key = fields.Char(copy=False) # used to resolve multiple assets in a multi-website environment
     website_id = fields.Many2one('website', ondelete='cascade')

--- a/addons/website/models/ir_attachment.py
+++ b/addons/website/models/ir_attachment.py
@@ -6,8 +6,7 @@ _logger = logging.getLogger(__name__)
 
 
 class IrAttachment(models.Model):
-
-    _inherit = ["ir.attachment"]
+    _inherit = "ir.attachment"
 
     # Technical field used to resolve multiple attachments in a multi-website environment.
     key = fields.Char()

--- a/addons/website/models/ir_binary.py
+++ b/addons/website/models/ir_binary.py
@@ -2,7 +2,7 @@ from odoo import models
 
 
 class IrBinary(models.AbstractModel):
-    _inherit = ['ir.binary']
+    _inherit = 'ir.binary'
 
     def _find_record(
             self, xmlid=None, res_model='ir.attachment', res_id=None,

--- a/addons/website/models/ir_http.py
+++ b/addons/website/models/ir_http.py
@@ -59,7 +59,7 @@ def get_request_website():
 
 
 class IrHttp(models.AbstractModel):
-    _inherit = ['ir.http']
+    _inherit = 'ir.http'
 
     def routing_map(self, key=None):
         if not key and request:

--- a/addons/website/models/ir_model.py
+++ b/addons/website/models/ir_model.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class Base(models.AbstractModel):
-    _inherit = ['base']
+    _inherit = 'base'
 
     def get_base_url(self):
         """

--- a/addons/website/models/ir_model_data.py
+++ b/addons/website/models/ir_model_data.py
@@ -9,7 +9,7 @@ _logger = logging.getLogger(__name__)
 
 
 class IrModelData(models.Model):
-    _inherit = ['ir.model.data']
+    _inherit = 'ir.model.data'
 
     @api.model
     def _process_end_unlink_record(self, record):

--- a/addons/website/models/ir_module_module.py
+++ b/addons/website/models/ir_module_module.py
@@ -14,6 +14,7 @@ _logger = logging.getLogger(__name__)
 
 
 class IrModuleModule(models.Model):
+    _name = 'ir.module.module'
     _description = 'Module'
     _inherit = ['ir.module.module']
 

--- a/addons/website/models/ir_qweb_fields.py
+++ b/addons/website/models/ir_qweb_fields.py
@@ -8,7 +8,7 @@ from odoo.addons.website.tools import add_form_signature
 
 
 class IrQwebFieldContact(models.AbstractModel):
-    _inherit = ['ir.qweb.field.contact']
+    _inherit = 'ir.qweb.field.contact'
 
     @api.model
     def get_available_options(self):
@@ -22,7 +22,7 @@ class IrQwebFieldContact(models.AbstractModel):
 
 
 class IrQwebFieldHtml(models.AbstractModel):
-    _inherit = ['ir.qweb.field.html']
+    _inherit = 'ir.qweb.field.html'
 
     @api.model
     def value_to_html(self, value, options):

--- a/addons/website/models/ir_rule.py
+++ b/addons/website/models/ir_rule.py
@@ -3,7 +3,7 @@ from odoo.addons.website.models import ir_http
 
 
 class IrRule(models.Model):
-    _inherit = ['ir.rule']
+    _inherit = 'ir.rule'
 
     @api.model
     def _eval_context(self):

--- a/addons/website/models/ir_ui_menu.py
+++ b/addons/website/models/ir_ui_menu.py
@@ -5,7 +5,7 @@ from odoo.http import request
 
 
 class IrUiMenu(models.Model):
-    _inherit = ['ir.ui.menu']
+    _inherit = 'ir.ui.menu'
 
     @api.model
     @tools.ormcache_context('self._uid', keys=('lang', 'force_action',))

--- a/addons/website/models/ir_ui_view.py
+++ b/addons/website/models/ir_ui_view.py
@@ -15,6 +15,7 @@ _logger = logging.getLogger(__name__)
 
 
 class IrUiView(models.Model):
+    _name = 'ir.ui.view'
 
     _inherit = ["ir.ui.view", "website.seo.metadata"]
 

--- a/addons/website/models/mixins.py
+++ b/addons/website/models/mixins.py
@@ -17,6 +17,7 @@ logger = logging.getLogger(__name__)
 
 
 class WebsiteSeoMetadata(models.AbstractModel):
+    _name = 'website.seo.metadata'
 
     _description = 'SEO metadata'
 
@@ -98,6 +99,7 @@ class WebsiteSeoMetadata(models.AbstractModel):
 
 
 class WebsiteCover_PropertiesMixin(models.AbstractModel):
+    _name = 'website.cover_properties.mixin'
 
     _description = 'Cover Properties Website Mixin'
 
@@ -152,6 +154,7 @@ class WebsiteCover_PropertiesMixin(models.AbstractModel):
 
 
 class WebsiteMultiMixin(models.AbstractModel):
+    _name = 'website.multi.mixin'
 
     _description = 'Multi Website Mixin'
 
@@ -173,6 +176,7 @@ class WebsiteMultiMixin(models.AbstractModel):
 
 
 class WebsitePublishedMixin(models.AbstractModel):
+    _name = 'website.published.mixin'
 
     _description = 'Website Published Mixin'
 

--- a/addons/website/models/res_company.py
+++ b/addons/website/models/res_company.py
@@ -5,7 +5,7 @@ from odoo.exceptions import ValidationError
 
 
 class ResCompany(models.Model):
-    _inherit = ["res.company"]
+    _inherit = "res.company"
 
     website_id = fields.Many2one('website', compute='_compute_website_id', store=True)
 

--- a/addons/website/models/res_config_settings.py
+++ b/addons/website/models/res_config_settings.py
@@ -9,7 +9,7 @@ from werkzeug import urls
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     def _default_website(self):
         return self.env['website'].search([('company_id', '=', self.env.company.id)], limit=1)

--- a/addons/website/models/res_lang.py
+++ b/addons/website/models/res_lang.py
@@ -7,7 +7,7 @@ from odoo.http import request
 
 
 class ResLang(models.Model):
-    _inherit = ["res.lang"]
+    _inherit = "res.lang"
 
     def write(self, vals):
         if 'active' in vals and not vals['active']:

--- a/addons/website/models/res_partner.py
+++ b/addons/website/models/res_partner.py
@@ -6,6 +6,7 @@ from odoo import models, fields, api
 
 
 class ResPartner(models.Model):
+    _name = 'res.partner'
     _inherit = ['res.partner', 'website.published.multi.mixin']
 
     visitor_ids = fields.One2many('website.visitor', 'partner_id', string='Visitors')

--- a/addons/website/models/res_users.py
+++ b/addons/website/models/res_users.py
@@ -9,7 +9,7 @@ _logger = logging.getLogger(__name__)
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     website_id = fields.Many2one('website', related='partner_id.website_id', store=True, related_sudo=False, readonly=False)
 

--- a/addons/website/models/theme_models.py
+++ b/addons/website/models/theme_models.py
@@ -12,6 +12,7 @@ _logger = logging.getLogger(__name__)
 
 
 class ThemeIrAsset(models.Model):
+    _name = 'theme.ir.asset'
     _description = 'Theme Asset'
 
     key = fields.Char()
@@ -49,6 +50,7 @@ class ThemeIrAsset(models.Model):
 
 
 class ThemeIrUiView(models.Model):
+    _name = 'theme.ir.ui.view'
     _description = 'Theme UI View'
 
     def compute_arch_fs(self):
@@ -108,6 +110,7 @@ class ThemeIrUiView(models.Model):
 
 
 class ThemeIrAttachment(models.Model):
+    _name = 'theme.ir.attachment'
     _description = 'Theme Attachments'
 
     name = fields.Char(required=True)
@@ -131,6 +134,7 @@ class ThemeIrAttachment(models.Model):
 
 
 class ThemeWebsiteMenu(models.Model):
+    _name = 'theme.website.menu'
     _description = 'Website Theme Menu'
 
     name = fields.Char(required=True, translate=True)
@@ -171,6 +175,7 @@ class ThemeWebsiteMenu(models.Model):
 
 
 class ThemeWebsitePage(models.Model):
+    _name = 'theme.website.page'
     _description = 'Website Theme Page'
 
     url = fields.Char()
@@ -210,6 +215,7 @@ class ThemeWebsitePage(models.Model):
 
 
 class ThemeUtils(models.AbstractModel):
+    _name = 'theme.utils'
     _description = 'Theme Utils'
     _auto = False
 
@@ -348,7 +354,7 @@ class ThemeUtils(models.AbstractModel):
 
 
 class IrUiView(models.Model):
-    _inherit = ['ir.ui.view']
+    _inherit = 'ir.ui.view'
 
     theme_template_id = fields.Many2one('theme.ir.ui.view', copy=False)
 
@@ -375,25 +381,25 @@ class IrUiView(models.Model):
 
 
 class IrAsset(models.Model):
-    _inherit = ['ir.asset']
+    _inherit = 'ir.asset'
 
     theme_template_id = fields.Many2one('theme.ir.asset', copy=False)
 
 
 class IrAttachment(models.Model):
-    _inherit = ['ir.attachment']
+    _inherit = 'ir.attachment'
 
     key = fields.Char(copy=False)
     theme_template_id = fields.Many2one('theme.ir.attachment', copy=False)
 
 
 class WebsiteMenu(models.Model):
-    _inherit = ['website.menu']
+    _inherit = 'website.menu'
 
     theme_template_id = fields.Many2one('theme.website.menu', copy=False)
 
 
 class WebsitePage(models.Model):
-    _inherit = ['website.page']
+    _inherit = 'website.page'
 
     theme_template_id = fields.Many2one('theme.website.page', copy=False)

--- a/addons/website/models/website.py
+++ b/addons/website/models/website.py
@@ -91,6 +91,7 @@ DEFAULT_BLOCKED_THIRD_PARTY_DOMAINS = '\n'.join([  # noqa: FLY002
 
 
 class Website(models.Model):
+    _name = 'website'
 
     _description = "Website"
     _order = "sequence, id"

--- a/addons/website/models/website_configurator_feature.py
+++ b/addons/website/models/website_configurator_feature.py
@@ -7,6 +7,7 @@ from odoo.exceptions import ValidationError
 
 
 class WebsiteConfiguratorFeature(models.Model):
+    _name = 'website.configurator.feature'
 
     _description = 'Website Configurator Feature'
     _order = 'sequence'

--- a/addons/website/models/website_controller_page.py
+++ b/addons/website/models/website_controller_page.py
@@ -4,6 +4,7 @@ from odoo import api, fields, models
 
 
 class WebsiteControllerPage(models.Model):
+    _name = 'website.controller.page'
     _inherits = {'ir.ui.view': 'view_id'}
     _inherit = [
         'website.published.multi.mixin',

--- a/addons/website/models/website_form.py
+++ b/addons/website/models/website_form.py
@@ -8,7 +8,7 @@ from odoo.osv import expression
 
 
 class Website(models.Model):
-    _inherit = ['website']
+    _inherit = 'website'
 
     def _website_form_last_record(self):
         if request and request.session.get('form_builder_model_model'):
@@ -17,6 +17,7 @@ class Website(models.Model):
 
 
 class IrModel(models.Model):
+    _name = 'ir.model'
     _description = 'Models'
     _inherit = ['ir.model']
 

--- a/addons/website/models/website_menu.py
+++ b/addons/website/models/website_menu.py
@@ -13,6 +13,7 @@ from odoo.tools.translate import html_translate
 
 
 class WebsiteMenu(models.Model):
+    _name = 'website.menu'
 
     _description = "Website Menu"
 

--- a/addons/website/models/website_page.py
+++ b/addons/website/models/website_page.py
@@ -10,6 +10,7 @@ from odoo.tools.translate import _
 
 
 class WebsitePage(models.Model):
+    _name = 'website.page'
     _inherits = {'ir.ui.view': 'view_id'}
     _inherit = [
         'website.published.multi.mixin',

--- a/addons/website/models/website_page_properties.py
+++ b/addons/website/models/website_page_properties.py
@@ -4,6 +4,7 @@ from odoo import api, fields, models
 
 
 class WebsitePagePropertiesBase(models.TransientModel):
+    _name = 'website.page.properties.base'
     _description = "Page Properties Base"
 
     target_model_id = fields.Reference(selection='_selection_target_model_id', required=True)
@@ -128,6 +129,7 @@ class WebsitePagePropertiesBase(models.TransientModel):
 
 
 class WebsitePageProperties(models.TransientModel):
+    _name = 'website.page.properties'
     _description = "Page Properties"
     _inherit = [
         'website.page.properties.base',

--- a/addons/website/models/website_rewrite.py
+++ b/addons/website/models/website_rewrite.py
@@ -11,6 +11,7 @@ _logger = logging.getLogger(__name__)
 
 
 class WebsiteRoute(models.Model):
+    _name = 'website.route'
     _rec_name = 'path'
     _description = "All Website Route"
     _order = 'path'
@@ -48,6 +49,7 @@ class WebsiteRoute(models.Model):
 
 
 class WebsiteRewrite(models.Model):
+    _name = 'website.rewrite'
     _description = "Website rewrite"
 
     name = fields.Char('Name', required=True)

--- a/addons/website/models/website_snippet_filter.py
+++ b/addons/website/models/website_snippet_filter.py
@@ -12,6 +12,7 @@ _logger = logging.getLogger(__name__)
 
 
 class WebsiteSnippetFilter(models.Model):
+    _name = 'website.snippet.filter'
     _inherit = ['website.published.multi.mixin']
     _description = 'Website Snippet Filter'
     _order = 'name ASC'

--- a/addons/website/models/website_visitor.py
+++ b/addons/website/models/website_visitor.py
@@ -16,6 +16,7 @@ from odoo.osv import expression
 
 
 class WebsiteTrack(models.Model):
+    _name = 'website.track'
     _description = 'Visited Pages'
     _order = 'visit_datetime DESC'
     _log_access = False
@@ -27,6 +28,7 @@ class WebsiteTrack(models.Model):
 
 
 class WebsiteVisitor(models.Model):
+    _name = 'website.visitor'
     _description = 'Website Visitor'
     _order = 'id DESC'
 
